apps/api/src/services/queues/data-gathering/data-gathering.processor.ts

```diff
@@ -142,12 +146,26 @@ export class DataGatheringProcessor {
         `DataGatheringProcessor (${GATHER_HISTORICAL_MARKET_DATA_PROCESS_JOB_NAME})`
       );
     } catch (error) {
+      if (error instanceof AssetProfileDelistedError) {
+        const updatedSymbolProfile: Prisma.SymbolProfileUpdateInput = {
+          isActive: false
+        };
+
+        await this.symbolProfileService.updateSymbolProfile(
+          {
+            dataSource,
+            symbol
+          },
+          updatedSymbolProfile
+        );
```

Comment:
What if we discard the job in case of delisting via
```
return job.discard();
```
to prevent further retries?